src/OpenColorIO/CPUProcessor.cpp

```diff
@@ -0,0 +1,1135 @@
+    {
+        // Step 1: Process the head of the list.
+
+        if(DynamicPtrCast<const Op>(ops[0])->data()->getType()==OpData::Lut1DType)
+        {
+            // Benefit from the 1D LUT look-up for integer bit depths.
+
+            if(ExtractChannelOrder(in)!=CHANNEL_ORDERING_RGBA)
+            {
+                m_ops.push_back( CreateChannelOrderOp(ExtractChannelOrder(in),
+                                                      CHANNEL_ORDERING_RGBA, 
+                                                      ExtractBitDepth(in)) );
+            }
+
+            m_ops.push_back( CreateCPUOp(ops[0], ExtractBitDepth(in), BIT_DEPTH_F32) );
+        }
+        else
+        {
+            if(in!=PIXEL_FORMAT_RGBA_F32)
+            {
+                m_ops.push_back( CreatePixelFormatOp(in, PIXEL_FORMAT_RGBA_F32) );
+            }
+
+            m_ops.push_back( CreateCPUOp(ops[0], BIT_DEPTH_F32, BIT_DEPTH_F32) );
+        }
+
+        // Step 2: Process the body of the list (i.e. all ops except the first and the last ones).
+
+        for(size_t idx=1; idx<(numOps-1); ++idx)
+        {
+            m_ops.push_back(ops[idx]->getCPUOp());
+        }
+
+        // Step 3: Process the tail of the list.
+
+        if(DynamicPtrCast<const Op>(ops[numOps-1])->data()->getType()==OpData::Lut1DType)
+        {
+            // Benefit from the 1D LUT native bit depth support.
+
+            m_ops.push_back( CreateCPUOp(ops[numOps-1], BIT_DEPTH_F32, ExtractBitDepth(out)) );
+
+            if(ExtractChannelOrder(out)!=CHANNEL_ORDERING_RGBA)
+            {
+                m_ops.push_back( CreateChannelOrderOp(CHANNEL_ORDERING_RGBA,
+                                                      ExtractChannelOrder(out), 
+                                                      ExtractBitDepth(out)) );
+            }
+        }
+        else
+        {
+            m_ops.push_back( CreateCPUOp(ops[numOps-1], BIT_DEPTH_F32, BIT_DEPTH_F32) );
+
+            if(out!=PIXEL_FORMAT_RGBA_F32)
+            {
+                // Convert from RGBA F32 to the output pixel format.
+                m_ops.push_back( CreatePixelFormatOp(PIXEL_FORMAT_RGBA_F32, out) );
+            }
+        }
+    }
+
+    // The optimization may result in an empty op list (e.g. a Processor is created 
+    // with the same src & dst colour space).
+
+    else if (in!=PIXEL_FORMAT_RGBA_F32 || out!=PIXEL_FORMAT_RGBA_F32)
+    {
+        // There is some conversion needed between the in and out pixel formats.
+
+        if(in!=PIXEL_FORMAT_RGBA_F32)
+        {
+            m_ops.push_back( CreatePixelFormatOp(in, PIXEL_FORMAT_RGBA_F32) );
+        }
+
+        if(out!=PIXEL_FORMAT_RGBA_F32)
+        {
+            m_ops.push_back( CreatePixelFormatOp(PIXEL_FORMAT_RGBA_F32, out) );
+        }
+    }
+    else
+    {
+        // There is some minimal processing needed to support different buffer images.
+
+        m_ops.push_back( CreatePixelFormatOp(PIXEL_FORMAT_RGBA_F32, PIXEL_FORMAT_RGBA_F32) );

```

Comment:
What processing is needed in this case?

Reply:
The input and output pixel formats could be different imposing the input and output image buffers to be different. Unfortunately, the finalization step could not guess anything about the buffer images, so it adds a 'NoOp' operator by default. And, the CPUProcessor must also succeed to be consistent with the OCIO v1 behavior.